src/Microsoft.OData.Core/MultipartMixed/ODataMultipartMixedBatchWriter.cs

```diff
@@ -0,0 +1,413 @@
+            // then dispose the batch writer (since we are now writing the operation content) and set the corresponding state.
+            return this.RawOutputContext.FlushBuffersAsync()
+                .FollowOnSuccessWith(task => this.DisposeBatchWriterAndSetContentStreamRequestedState());
+        }
+#endif
+
+        /// <summary>
+        /// This method is called to notify that the content stream of a batch operation has been disposed.
+        /// </summary>
+        public override void BatchOperationContentStreamDisposed()
+        {
+            Debug.Assert(this.CurrentOperationMessage != null, "Expected non-null operation message!");
+
+            this.SetState(BatchWriterState.OperationStreamDisposed);
+            this.CurrentOperationRequestMessage = null;
+            this.CurrentOperationResponseMessage = null;
+            this.RawOutputContext.InitializeRawValueWriter();
+        }
+
+        /// <summary>
+        /// This method notifies the listener, that an in-stream error is to be written.
+        /// </summary>
+        /// <remarks>
+        /// This listener can choose to fail, if the currently written payload doesn't support in-stream error at this position.
+        /// If the listener returns, the writer should not allow any more writing, since the in-stream error is the last thing in the payload.
+        /// </remarks>
+        public override void OnInStreamError()
+        {
+            this.RawOutputContext.VerifyNotDisposed();
+            this.SetState(BatchWriterState.Error);
+            this.RawOutputContext.TextWriter.Flush();
+
+            // The OData protocol spec did not defined the behavior when an exception is encountered outside of a batch operation. The batch writer
+            // should not allow WriteError in this case. Note that WCF DS Server does serialize the error in XML format when it encounters one outside of a
+            // batch operation.
+            throw new ODataException(Strings.ODataBatchWriter_CannotWriteInStreamErrorForBatch);
+        }
+
+        /// <summary>
+        /// Flush the output.
+        /// </summary>
+        protected override void FlushSynchronously()
+        {
+            this.RawOutputContext.Flush();
+        }
+
+#if PORTABLELIB
+        /// <summary>
+        /// Flush the output.
+        /// </summary>
+        /// <returns>Task representing the pending flush operation.</returns>
+        protected override Task FlushAsynchronously()
+        {
+            return this.RawOutputContext.FlushAsync();
+        }
+#endif
+
+        /// <summary>
+        /// Starts a new changeset - implementation of the actual functionality.
+        /// </summary>
+        /// <param name="changeSetGuid">The value for changeset boundary for multipart batch.</param>
+        protected override void WriteStartChangesetImplementation(string changeSetGuid)

```

Comment:
>changeSetGuid [](start = 73, length = 13)

Should not be called "changeSetGuid" -- it's not necessarily a Guid, it could be any string identifier. "changeSetId" would be more accurate.

Reply:
Renamed here and in ODataMultipartMixedBatchWriterUtils. Solution-wide search doesn't indicate other changes needed.

---
In reply to: [156769543](https://github.com/OData/odata.net/pull/992#discussion_r156769543) [](ancestors = 156769543)